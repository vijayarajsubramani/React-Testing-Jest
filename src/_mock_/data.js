export const userData = [
    {
        id: 1,
        first_name: 'vijayaraj',
        email: 'vijay@gmail.com',
        avatar: 'vijay.jpg'
    }
]

export const badResponse = {
    error: 'error',
    title: 'something went wrong'
}