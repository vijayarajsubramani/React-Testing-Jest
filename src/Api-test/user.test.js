import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import User from './users'
// import { rest } from 'msw'
// import { setupServer } from 'msw/node'
import { userData } from '../_mock_/data'

// const server = setupServer(
//     rest.get('"https://reqres.in/api/users/', (req, res, ctx) => {
//         return res(ctx.status(200), ctx.json(userData))
//     })
// )
describe('render the user component', () => {

    // beforeAll(() => server.listen());
    // afterEach(() => server.resetHandlers());
    // afterAll(() => server.close());

    test('render the header text as users', () => {
        const { container } = render(<User />)
        const elementheading = screen.getByRole('heading', { level: 1 })
        expect(container).toMatchSnapshot();
        expect(elementheading).toBeTruthy()
    })
    test('search bar input is empty', async () => {
        const { container } = render(<User />)
        const inputSearch = await waitFor(() => screen.getByPlaceholderText(/search user/i))
        expect(container).toMatchSnapshot();
        expect(inputSearch).toHaveAttribute("value", '')
    })
    test('search bar input has value', async () => {
        const { container } = render(<User />)
        const inputSearch = await waitFor(() => screen.getByPlaceholderText(/search user/i))
        fireEvent.change(inputSearch, { target: { value: 'tests' } })
        expect(container).toMatchSnapshot();
        expect(inputSearch).toHaveAttribute("value", 'tests')
    })
    // test('return json when response is ok', async () => {
    //     expect.assertions(3)
    //     const res = await fetch('https://reqres.in/api/users/', { method: 'GET' })
    //     const data = await res.json()
    //     expect(data.status).toBe(200)
    //     expect(data.ok).toBe(true)
    //     expect(JSON.stringify(data)).toBe(JSON.stringify(userData))
    // })
})
