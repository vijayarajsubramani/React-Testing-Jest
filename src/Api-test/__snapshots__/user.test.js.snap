// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the user component render the header text as users 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Users 
    </h1>
    <input
      placeholder="search user"
      type="text"
      value=""
    />
    <div
      class="flex"
    />
  </div>
</div>
`;

exports[`render the user component search bar input has value 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Users 
    </h1>
    <input
      placeholder="search user"
      type="text"
      value="tests"
    />
    <div
      class="flex"
    />
  </div>
</div>
`;

exports[`render the user component search bar input is empty 1`] = `
<div>
  <div
    class="App"
  >
    <h1>
      Users 
    </h1>
    <input
      placeholder="search user"
      type="text"
      value=""
    />
    <div
      class="flex"
    />
  </div>
</div>
`;
